SWLOR.Game.Server/Service/HoloComService.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System;
+using SWLOR.Game.Server.GameObject;
+using NWN;
+using static NWN._;
+using SWLOR.Game.Server.Event.Module;
+using SWLOR.Game.Server.Messaging;
+using System.Linq;
+using SWLOR.Game.Server.Enumeration;
+using SWLOR.Game.Server.NWNX;
+
+namespace SWLOR.Game.Server.Service
+{
+    public static class HoloComService
+    {
+        /*
+         * 
+        -- retest -- Languages shouldn't get translated by holocom.
+        -- retest -- Need to test if you get language xp from your hologram.
+        -- retest -- calling someone who's busy borks you.
+        -- retest -- if player dies while in call they resurrect without cutscene immobile.
+        -- retest -- only allow talk, whisper and party channels for holograms. NO DM or SHOUT 
+        ??? can limbo holograms but they dont show in limbo.
+         * 
+         */
+
+        public static void SubscribeEvents()
+        {
+            MessageHub.Instance.Subscribe<OnModuleChat>(message => OnModuleChat());
+            MessageHub.Instance.Subscribe<OnModuleDeath>(message => OnModuleDeath());
+        }
+
+        private static void OnModuleDeath()
+        {
+            NWPlayer player = _.GetLastPlayerDied();
+            if (HoloComService.IsInCall(player)) HoloComService.SetIsInCall(player, HoloComService.GetTargetForActiveCall(player), false);
+
+        }
+
+        private static void OnModuleChat()
+        {
+            NWPlayer sender = GetPCChatSpeaker();
+            int talkvolume = GetPCChatVolume();
+
+            ChatChannelType channel = (ChatChannelType)NWNXChat.GetChannel();
+
+            if (!IsInCall(sender)) return;
+            if (channel != ChatChannelType.PlayerTalk) return;
+            if (channel != ChatChannelType.PlayerWhisper) return;
+            if (channel != ChatChannelType.PlayerParty) return;
+
+            //if (talkvolume == TALKVOLUME_SHOUT) return;
+
+            NWPlayer receiver = GetHoloGram(sender);
+
+            string text = _.GetPCChatMessage().Trim();
+
+            if (text.StartsWith("/")) return;
+
+            int animation = ANIMATION_LOOPING_TALK_NORMAL;
+            if (text.Contains("!")) animation = ANIMATION_LOOPING_TALK_FORCEFUL;
+            if (text.Contains("?")) animation = ANIMATION_LOOPING_TALK_PLEADING;
+
+            SetCommandable(TRUE, receiver);
+            receiver.ClearAllActions();
+
+            receiver.AssignCommand(() =>
+            {
+                ActionPlayAnimation(animation);
+            });
+
+            receiver.DelayAssignCommand(() =>
+            {
+                ActionSpeakString(text, talkvolume);
+            },0.3f);
+
+            /* animation stuff
```

Comment:
Probably want to leave this debug info out of the code base.

Reply:
I left it there so the future change and potential solution is documented somewhere. Should move it to an "Issue" instead?

Reply:
Yea let's move it to an issue and just tag the PR so we can reference it in the future. I just think it confuses things being in the code itself